### ISSUE TITLE
Can't Connect Using SSL

### ISSUE THREAD

**Issue by SimonBrazell** [zcoinofficial/ccminer#78]
Hello, I've been attempting to connect using SSL to `firo.mintpond.com` and have been unable to do so, connection works fine using TCP.

Launch arguments are - `ccminer_cuda11.exe -a mtp -u wallet.worker -p password -o stratum+ssl://firo.mintpond.com:3005`

This is met with the following output:
```
*** ccminer 1.3.2-djm34 for nVidia GPUs by djm34 ***
    Built with VC++ 2015 and nVidia CUDA SDK 11.1

  Originally based on Christian Buchner and Christian H. project based on tpruvot 1.8.4 release
  Include algos from alexis78, djm34, sp, tsiv and klausT.
  *** News (07/06/2018): MTP algo for ZCoin

  MTP algo based on krnlx kernel

  BTC donation address: 1NENYmxwZGHsKFmyjTc5WferTn5VTFb7Ze (djm34)
  ZCoin donation address: aChWVb8CpgajadpLmiwDZvZaKizQgHxfh5 (djm34)

Try `ccminer --help' for more information.
```

Works fine with - `ccminer_cuda11.exe -a mtp -u wallet.worker -p password -o stratum+tcp://firo.mintpond.com:3000`

I confirmed the SSL certificate is current.

**Comment by djm34** [comment]
As written in the miner, when you type a wrong command: 
"Try `ccminer --help' for more information."

there you'll find (small excerpt): 

  -i  --intensity=N[,N] GPU intensity 8.0-25.0 (default: auto)
                        Decimals are allowed for fine tuning
      --cuda-schedule   Set device threads scheduling mode (default: auto)
  -f, --diff-factor     Divide difficulty by this factor (default 1.0)
  -m, --diff-multiplier Multiply difficulty by this value (default 1.0)
  -o, --url=URL         URL of mining server
  -O, --userpass=U:P    username:password pair for mining server
  -u, --user=USERNAME   username for mining server
  -p, --pass=PASSWORD   password for mining server
      --cert=FILE       certificate for mining server using SSL 

so basically you need to use --cert="your certifcate" with stratum+tcp (stratum+ssl is a command which doesn't exist)

**Comment by SimonBrazell** [comment]
Okay thanks @djm34 much appreciated, I'll give that a go now instead.

**Comment by SimonBrazell** [comment]
Okay so I attempted the following:
```
ccminer_cuda11.exe -a mtp -u wallet.worker -p password --cert="certificate.crt" -o stratum+tcp://firo-us.mintpond.com:3005
```

Where `certificate.crt` was obtained by running:
```
echo | openssl s_client -servername firo-us.mintpond.com -connect firo-us.mintpond.com:3005 |\
  sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' > certificate.crt
```
But it failed to connect, would appreciate some guidance 🙂

I'm told that SSL isn't really necessary by some other miners in the pool but I'm more just curious now.

**Comment by djm34** [comment]
to be honest I never used it, so I can't give you much help

**Comment by SimonBrazell** [comment]
Okay no worries, I guess I'll leave it at that, I guess no one else uses it either.

It seemed weird to me that I had to provide a certificate as the client anyway, I'll let you know if I come across anything in future though.

**Comment by djm34** [comment]
looking into the code, it seems that the use of the cerificate is limited to solo mining (and for the old "getwork" method), but it doesn't seem there is anything for pool. Looking into mintpond FAQ beside the ssl/tls port they don't say anything either.

**Comment by SimonBrazell** [comment]
Yeah I spoke to JCTheMiner on the mintpond discord and they weren't sure that ccminer was compatible with the TLS endpoint. There is [a page](https://mintpond.com/b/how-to-mine-firo) on mintpond's site that produces a config for the miner and endpoint you choose, that's where I originally got the `stratum+ssl` arg I posted above.
